components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/utils/AppManagerUtil.java

```diff
@@ -1167,21 +1141,25 @@ public static String createEndpoint(String endpointUrl, Registry registry)
 					Tier tier = new Tier(id.getText());
 					tier.setPolicyContent(policy.toString().getBytes());
 					tier.setDisplayName(displayName);
-					// String desc =
-					// resource.getProperty(AppMConstants.TIER_DESCRIPTION_PREFIX
-					// + id.getText());
-					String desc;
+
 					try {
 						desc = APIDescriptionGenUtil.generateDescriptionFromPolicy(policy);
 					} catch (AppManagementException ex) {
 						desc = AppMConstants.TIER_DESC_NOT_AVAILABLE;
 					}
+					try {
+						tierMaxCount = APIDescriptionGenUtil.generateMaxCountFromPolicy(policy);
+					} catch (AppManagementException ex) {
```

Comment:
If the exception occurred when getting a max count from policy we will log it inside the generateMaxCountFromPolicy method.